python/hsfs/core/vector_db_client.py

```diff
@@ -185,14 +191,32 @@ def find_neighbors(
         return [
             (
                 1 / item["_score"] - 1,
-                self._rewrite_result_key(
+                self._convert_to_pandas_type(schema, self._rewrite_result_key(
                     item["_source"],
                     self._fg_vdb_col_td_col_map[embedding_feature.feature_group.id],
-                ),
+                )),
             )
             for item in results["hits"]["hits"]
         ]
 
+    def _convert_to_pandas_type(self, schema, result):
+        for feature in schema:
+            feature_name = feature.name
+            feature_type = feature.type.lower()
+            feature_value = result.get(feature_name)
+            if not feature_value:  # Feature value can be null
+                continue
+            elif feature_type == "date":
+                result[feature_name] = datetime.utcfromtimestamp(feature_value // 10**3).date()
+            elif feature_type == "timestamp":
+                # convert timestamp in ms to datetime in s
+                result[feature_name] = datetime.utcfromtimestamp(feature_value // 10**3)
+            elif feature_type == "binary":
+                result[feature_name] = base64.b64decode(feature_value)
+            elif feature.is_complex() and feature not in self._embedding_features:
+                result[feature_name] = base64.b64decode(feature_value)
```

Comment:
to avoid duplicating: `result[feature_name] = base64.b64decode(feature_value)`
```suggestion
            elif feature_type == "binary" or (feature.is_complex() and feature not in self._embedding_features):
                result[feature_name] = base64.b64decode(feature_value)
```
also complex feature are `"binary"` so is `feature.is_complex() and feature not in self._embedding_features` needed?

Reply:
@bubriks modified. here `feature_type` is the offline feature type, so they are either array or strct.